manual/src/cmds/intf-c.etex

```diff
@@ -2500,9 +2500,11 @@ CAMLprim stub_gethostbyname(value vname)
 }
 \end{verbatim}
 
-During the time the domain lock is released, the thread-local variable
-"Caml_state" is set to "NULL". This can be used to determine if the
-thread currently holds its domain lock.
+The thread-local variable "Caml_state_opt" is set to "NULL" when the
+domain lock is not held, and contains a pointer to the domain-local
+state otherwise. This can be used to determine whether the thread
+currently holds its domain lock. "Caml_state" is equal to
+"Caml_state_opt", with a check in debug mode that it is not "NULL".
```

Comment:
Same comment as above: I think it's clearer if we describe the dynamic-error-check behavior of `Caml_state` and introduce `Caml_state_opt`, the lower-level mechanism, second.

Reply:
Thanks!